static/js/src/advantage/react/components/Subscriptions/ExpiryNotification/ExpiryNotification.test.tsx

```diff
@@ -0,0 +1,107 @@
+import React from "react";
+import { mount } from "enzyme";
+
+import ExpiryNotification, {
+  ExpiryNotificationSize,
+} from "./ExpiryNotification";
+import { userSubscriptionStatusesFactory } from "advantage/tests/factories/api";
+import { Notification } from "@canonical/react-components";
+
+describe("ExpiryNotification", () => {
+  it("can display an expiring notification", () => {
+    const statuses = userSubscriptionStatusesFactory.build({
+      is_expiring: true,
+    });
+    const wrapper = mount(
+      <ExpiryNotification
+        size={ExpiryNotificationSize.Large}
+        statuses={statuses}
+      />
+    );
+    expect(wrapper.find("[data-test='is_expiring-large']").exists()).toBe(true);
+  });
+
+  it("can display a large expired notification", () => {
+    const statuses = userSubscriptionStatusesFactory.build({
+      is_expired: true,
+    });
+    const wrapper = mount(
+      <ExpiryNotification
+        size={ExpiryNotificationSize.Large}
+        statuses={statuses}
+      />
+    );
+    expect(wrapper.find("[data-test='is_expired-large']").exists()).toBe(true);
+  });
+
+  it("can display a grace-period notification", () => {
+    const statuses = userSubscriptionStatusesFactory.build({
+      is_in_grace_period: true,
+    });
+    const wrapper = mount(
+      <ExpiryNotification
+        size={ExpiryNotificationSize.Large}
+        statuses={statuses}
+      />
+    );
+    expect(
+      wrapper.find("[data-test='is_in_grace_period-large']").exists()
+    ).toBe(true);
+  });
+
+  it("can display multiple notifications", () => {
+    const statuses = userSubscriptionStatusesFactory.build({
+      is_in_grace_period: true,
+      is_expired: true,
+      is_expiring: false,
+    });
+    const wrapper = mount(
+      <ExpiryNotification
+        multiple
+        size={ExpiryNotificationSize.Large}
+        statuses={statuses}
+      />
+    );
+    expect(
+      wrapper.find("[data-test='is_in_grace_period-large']").exists()
+    ).toBe(true);
+    expect(wrapper.find("[data-test='is_expired-large']").exists()).toBe(true);
+    expect(wrapper.find("[data-test='is_expiring-large']").exists()).toBe(
+      false
+    );
+  });
+
+  it("shows the highest priority notification when only showing one", () => {
+    const statuses = userSubscriptionStatusesFactory.build({
+      is_in_grace_period: true,
+      is_expired: true,
```

Comment:
I noticed that you use snake case for lots of properties which seems inconsistent with how most JS code is structured in this repo (at least in advantage camelCase seems to dominate).

Is snake_case the direction we'd like to move towards now? It might be a good idea to ask other members of the team how they feel about the change (unless such conversation already happened!).

Reply:
These values are from the API, they're in snake case and we don't transform the values to/from camel case as we interact with the API.

In this case the factory is building an API response (so even if we did transform to/from camel case the factory would still use snake case to mimic the API).

In the past we had some projects that transformed the case for the API values but we've found it much easier and less error prone to work with the data as it's sent/received by the API.

There's no intention to use snake case for anything other than API data.